src/components/App/App.jsx

```diff
@@ -122,43 +172,141 @@ function App() {
     }
   };
 
-  // Set a new recipe to recipes list array
+  // Set a new recipeto server then to recipes list array
   const handleRecipeSubmit = (item) => {
```

Comment:
You can create a special variable for watching **loading** states during server requests:

```js
 const [isLoading, setIsLoading] = React.useState(false);

```

And then you can change it to `true` before sending requests and return it back to `false`  in `finally` to stop loading:

And you should pass `isLoading` to every popup to change the **button text**:

```js

 buttonText={isLoading? 'Saving...' : 'Save'}

```